src/index.tsx

```diff
@@ -130,23 +155,28 @@ function LazyHydrate(props: Props) {
       }
     }
 
-    events.forEach(event => {
-      childRef.current.addEventListener(event, hydrate, {
-        once: true,
-        capture: true,
-        passive: true
-      });
-      cleanupFns.push(() => {
-        childRef.current.removeEventListener(event, hydrate, { capture: true });
+    const elToListenOn = listenOnEl || childRef.current;
+    if (elToListenOn) {
+      events.forEach(event => {
+        elToListenOn.addEventListener(event, hydrate, {
+          once: true,
+          capture: true,
+          passive: true
+        });
+        cleanupFns.push(() => {
+          elToListenOn.removeEventListener(event, hydrate, {
+            capture: true
+          });
+        });
       });
-    });
+    }
 
     return cleanup;
   }, [hydrated, on, ssrOnly, whenIdle, whenVisible, didHydrate, promise]);
 
   if (hydrated) {
     if (noWrapper) {
-      return children;
+      return <>{children}</>;
```

Comment:
no fragment needed

Reply:
In the case of noWrapper, since children could be an array, and an array is not a valid return type for a component, we should wrap them in a fragment `<>{children}</>`